include/deal.II/dofs/dof_handler_policy.h

```diff
@@ -69,7 +71,8 @@ namespace internal
          */
         virtual
         NumberCache
-        distribute_dofs (dealii::DoFHandler<dim,spacedim> &dof_handler) const = 0;
+        distribute_dofs (dealii::DoFHandler<dim,spacedim> &dof_handler,
+                         NumberCache &number_cache_current) const = 0;
```

Comment:
I think I still don't get it -- why return the same value both through the reference argument *and* the return value? Wouldn't it be simpler to just take it as a const argument and return the modification? The caller can then assign the result to the old object if they want.